pennylane/devices/default_clifford.py

```diff
@@ -0,0 +1,707 @@
+from pennylane.transforms.core import TransformProgram
+from pennylane.measurements.expval import ExpectationMP
+from pennylane.devices.qubit.sampling import get_num_shots_and_executions
+from pennylane.devices.qubit.simulate import INTERFACE_TO_LIKE
+
+from . import Device
+from .execution_config import ExecutionConfig, DefaultExecutionConfig
+
+from .default_qubit import accepted_sample_measurement
+
+from .preprocess import (
+    validate_observables,
+    validate_measurements,
+    validate_multiprocessing_workers,
+    validate_device_wires,
+    warn_about_trainable_observables,
+)
+
+Result_or_ResultBatch = Union[Result, ResultBatch]
+QuantumTapeBatch = Sequence[QuantumTape]
+QuantumTape_or_Batch = Union[QuantumTape, QuantumTapeBatch]
+# always a function from a resultbatch to either a result or a result batch
+PostprocessingFn = Callable[[ResultBatch], Result_or_ResultBatch]
+
+# Updated observable list
+_MEAS_OBSERVABLES = {
+    "PauliX",
+    "PauliY",
+    "PauliZ",
+    "Hadamard",
+    "Hermitian",
+    "Identity",
+    "Projector",
+    "Hamiltonian",
+    "Sum",
+    "SProd",
+    "Prod",
+    "Exp",
+}
+
+# Clifford gates
+_GATE_OPERATIONS = {
+    "Identity": "I",
+    "Snapshot": None,
+    "BasisState": None,
+    "StatePrep": None,
+    "PauliX": "X",
+    "PauliY": "Y",
+    "PauliZ": "Z",
+    "Hadamard": "H",
+    "S": "S",
+    "Adjoint(S)": "S_DAG",
+    "SX": "SX",
+    "Adjoint(SX)": "SX_DAG",
+    "CNOT": "CNOT",
+    "SWAP": "SWAP",
+    "ISWAP": "ISWAP",
+    "Adjoint(ISWAP)": "ISWAP_DAG",
+    "CY": "CY",
+    "CZ": "CZ",
+    "GlobalPhase": None,
+    "WireCut": None,
+}
+
+
+def observable_stopping_condition(obs: qml.operation.Operator) -> bool:
+    """Specifies whether or not an observable is accepted by DefaultClifford."""
+    return obs.name in _MEAS_OBSERVABLES
+
+
+def _import_stim():
+    """Import stim."""
+    try:
+        # pylint: disable=import-outside-toplevel, unused-import, multiple-imports
+        import stim
+    except ImportError as Error:
+        raise ImportError(
+            "This feature requires stim, a fast stabilizer circuit simulator."
+            "It can be installed with: pip install stim."
+        ) from Error
+    return stim
+
+
+class DefaultClifford(Device):
+    r"""A PennyLane device written in Python and capable of executing Clifford circuits using `stim (2021) <https://github.com/quantumlib/stim/tree/main>`_  .
+
+    Args:
+        shots (int, Sequence[int], Sequence[Union[int, Sequence[int]]]): The default number of shots to use in executions involving
+            this device.
+        check_clifford (bool): Check if all the gate operations in the circuits to be executed are Clifford. Default is ``True``.
+        max_error (float): The maximum permissible operator norm error for decomposing circuits with non-Clifford gate operations
+            into Cliffordt+T basis. The default is ``0.0`` as this device currently supports only Clifford simulations.
```

Comment:
ahh, not the explanation I was expecting. We _might_ want to set a different default value, otherwise we'll have to change it in the future, or we'll have to change what we mean by zero

Reply:
Made it `None` and explained it with the context of just supporting Clifford operations for now.